pkg/operators/kubeipresolver/kubeipresolver.go

```diff
@@ -0,0 +1,306 @@
+	if err != nil {
+		return nil, fmt.Errorf("creating new k8s clientset: %w", err)
+	}
+
+	return &k8sInventoryCache{
+		clientset:      clientset,
+		tickerDuration: tickerDuration,
+	}, nil
+}
+
+func (cache *k8sInventoryCache) loop() {
+	for {
+		select {
+		case <-cache.exit:
+			return
+		case <-cache.ticker.C:
+			cache.update()
+		}
+	}
+}
+
+func (cache *k8sInventoryCache) Close() {
+	if cache.exit != nil {
+		close(cache.exit)
+	}
+	if cache.ticker != nil {
+		cache.ticker.Stop()
+	}
+}
+
+func (cache *k8sInventoryCache) Start() {
+	cache.useCountMutex.Lock()
+	defer cache.useCountMutex.Unlock()
+
+	// No uses before us, we are the first one
+	if cache.useCount == 0 {
+		cache.update()
+		cache.exit = make(chan struct{})
+		cache.ticker = time.NewTicker(cache.tickerDuration)
+		go cache.loop()
+	}
+	cache.useCount++
+}
+
+func (cache *k8sInventoryCache) Stop() {
+	cache.useCountMutex.Lock()
+	defer cache.useCountMutex.Unlock()
+
+	// We are the last user, stop everything
+	if cache.useCount == 1 {
+		cache.Close()
+	}
+	cache.useCount--
+}
+
+func (cache *k8sInventoryCache) update() {
+	pods, err := cache.clientset.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		log.Errorf("listing pods: %v", err)
+		return
+	}
+	cache.pods.Store(pods)
+
+	svcs, err := cache.clientset.CoreV1().Services("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		log.Errorf("listing services: %s", err)
+		return
+	}
+	cache.svcs.Store(svcs)
+}
+
+func (cache *k8sInventoryCache) GetPods() *v1.PodList {
+	return cache.pods.Load()
+}
+
+func (cache *k8sInventoryCache) GetSvcs() *v1.ServiceList {
+	return cache.svcs.Load()
+}
+
+type KubeIPResolver struct {
+	k8sInventory *k8sInventoryCache
+}
+
+func (k *KubeIPResolver) Name() string {
+	return OperatorName
+}
+
+func (k *KubeIPResolver) Description() string {
+	return "KubeIPResolver resolves IP addresses to pod and service names"
+}
+
+func (k *KubeIPResolver) GlobalParamDescs() params.ParamDescs {
+	return nil
+}
+
+func (k *KubeIPResolver) ParamDescs() params.ParamDescs {
+	return nil
+}
+
+func (k *KubeIPResolver) Dependencies() []string {
+	return []string{"KubeManager"}
+}
+
+func (k *KubeIPResolver) CanOperateOn(gadget gadgets.GadgetDesc) bool {
+	km := kubemanager.KubeManager{}
+	if !km.CanOperateOn(gadget) {
+		return false
+	}
+
+	_, hasAdditionalInf := gadget.EventPrototype().(KubeNetworkInformation)
+	return hasAdditionalInf
+}
+
+func (k *KubeIPResolver) Init(params *params.Params) error {
+	k8sInventory, err := newCache(1 * time.Second)
+	if err != nil {
+		return fmt.Errorf("creating new k8sInventoryCache: %w", err)
+	}
+	k.k8sInventory = k8sInventory
+	return nil
+}
+
+func (k *KubeIPResolver) Close() error {
+	k.k8sInventory.Close()
+	return nil
+}
+
+func (k *KubeIPResolver) Instantiate(gadgetCtx operators.GadgetContext, gadgetInstance any, params *params.Params) (operators.OperatorInstance, error) {
+	traceInstance := &KubeIPResolverInstance{
```

Comment:
traceInstance -> operatorInstance? 

Or avoid this temp variable and return directly:

	return &KubeIPResolverInstance{
		gadgetCtx:      gadgetCtx,
		manager:        k,
		gadgetInstance: gadgetInstance,
	}, nil

Reply:
I noticed the same and it is the same for other operators. I would change it in all of them.